packages/xo-web/src/xo-app/hub/resource-form.js

```diff
@@ -3,38 +3,65 @@ import _ from 'intl'
 import decorate from 'apply-decorators'
 import Icon from 'icon'
 import React from 'react'
+import SingleLineRow from 'single-line-row'
 import Tooltip from 'tooltip'
-import { Container } from 'grid'
-import { SelectPool } from 'select-objects'
-import { error } from 'notification'
+import { Container, Col } from 'grid'
+import { find, first } from 'lodash'
+import { SelectPool, SelectSr } from 'select-objects'
+import { isSrWritable } from 'xo'
 import { injectState, provideState } from 'reaclette'
 
 export default decorate([
   provideState({
     initialState: ({ multi }) => ({
       pools: multi ? [] : undefined,
+      mapPoolsSrs: {},
     }),
     effects: {
       onChangePool(__, pools) {
-        const noDefaultSr = Array.isArray(pools)
-          ? pools.some(pool => pool.default_SR === undefined)
-          : pools.default_SR === undefined
-        if (noDefaultSr) {
-          error(_('hubSrErrorTitle'), _('noDefaultSr'))
-        } else {
-          this.props.onChange({
-            pools,
-            pool: pools,
-          })
-          return {
-            pools,
+        this.props.onChange({
+          pools,
+          mapPoolsSrs: this.state.mapPoolsSrs,
+        })
+        const _mapPoolsSrs = {}
+        if (Array.isArray(pools)) {
+          for (const pool of pools) {
+            _mapPoolsSrs[pool.id] = pool.default_SR
           }
+        } else {
+          _mapPoolsSrs[pools.id] = pools.default_SR
+        }
+        return {
+          pools,
+          mapPoolsSrs: _mapPoolsSrs,
```

Comment:
What if the user:
- selects a pool
- chooses a different SR than the default SR for that pool
- selects a second pool

Then the first pool's SR would be reset to the default SR?

Reply:
You're right, I fixed it using `lodash differenceBy` [code](https://github.com/vatesfr/xen-orchestra/pull/4571/files#diff-19f1a84d6035c2fa112b404861afeab9R25) thereby, only new selected pools will be set to default.